Services/Simulations.cs

```diff
@@ -228,9 +273,37 @@ public class Simulations : ISimulations
             return simulation;
         }
 
+        /// <summary>
+        /// Delete a simulation and its devices
+        /// </summary>
         public async Task DeleteAsync(string id)
         {
+            // Delete devices first
+            var deviceIds = this.GetDeviceIds(await this.GetAsync(id));
+            await this.devices.DeleteListAsync(deviceIds);
```

Comment:
If this fails do the devices stay in the IotHub?

Reply:
I'd be ok if they did as delete isn't used by anyone but us from postman right now.

Reply:
@jillcary yes, there is no error handling/retry here, because we use this endpoint only for tests/debugging right now. In future we'll probably promote this to be the official behavior, at which point we'll want to handle failures as you suggest